src/shared/pages/welcomePage/script.js

```diff
@@ -20,63 +20,45 @@ export async function checkPrivateBrowsing() {
   const alwaysPrivateCheckbox = document.getElementById(
     "always-private-checkbox",
   );
-  const currentExternalBrowser = await getExternalBrowser();
-  if (currentExternalBrowser === "Firefox Private Browsing") {
-    alwaysPrivateCheckbox.checked = true;
-  } else {
-    alwaysPrivateCheckbox.checked = false;
-  }
 
-  // On change, update the default launch mode
-  alwaysPrivateCheckbox.addEventListener("change", async () => {
+  alwaysPrivateCheckbox.checked =
+    (await getExternalBrowser()) === "Firefox Private Browsing";
+
+  browser.storage.sync.onChanged.addListener((changes) => {
+    if (changes.currentExternalBrowser !== undefined) {
+      alwaysPrivateCheckbox.checked =
+        changes.currentExternalBrowser.newValue === "Firefox Private Browsing";
+    }
+  });
+
+  alwaysPrivateCheckbox.addEventListener("click", async () => {
     let from = await getExternalBrowser();
     await handleChangeDefaultLaunchContextMenuClick();
     let to = await getExternalBrowser();
 
-    if (to === "Firefox Private Browsing") {
-      alwaysPrivateCheckbox.checked = true;
-    } else {
-      alwaysPrivateCheckbox.checked = false;
-    }
-
     settingEvent.currentBrowser.record({
       from,
       to,
       source: "always_private_checkbox",
     });
   });
-
-  browser.storage.sync.onChanged.addListener((changes) => {
-    if (changes.currentExternalBrowser !== undefined) {
-      if (
-        changes.currentExternalBrowser.newValue === "Firefox Private Browsing"
-      ) {
-        alwaysPrivateCheckbox.checked = true;
-      } else {
-        alwaysPrivateCheckbox.checked = false;
-      }
-    }
-  });
 }
 
 /**
  * Update the telemetry checkbox and initialize the listener.
  */
 export async function updateTelemetry() {
   // check storage to see if telemetry is enabled/disabled. If neither, false is default
-  const telemetryEnabled = await getTelemetryEnabled();
-  document.getElementById("telemetry-checkbox").checked = telemetryEnabled;
+  const telemetryCheckbox = document.getElementById("telemetry-checkbox");
+  telemetryCheckbox.checked = await getTelemetryEnabled();
 
-  document
-    .getElementById("telemetry-checkbox")
-    .addEventListener("change", (event) => {
-      browser.storage.sync.set({ telemetryEnabled: event.target.checked });
-    });
+  telemetryCheckbox.addEventListener("click", function () {
```

Comment:
Why "click" instead of a "change" listener? Change is semantically more correct here.